src/helpers.rs

```diff
@@ -210,3 +230,116 @@ pub trait Template: Serialize {
         Ok(tt.render("path", self)?)
     }
 }
+
+#[derive(Debug)]
+pub struct AsyncFileWriter {
+    /// Use AutoAbortJoinHandle so that the task
+    /// will be cancelled on failure.
+    handle: AutoAbortJoinHandle<io::Result<()>>,
+    tx: mpsc::Sender<Bytes>,
+}
```

Comment:
Might as well put those two in their own files, split up helpers before it sprawls too much

Reply:
I have the exact same thought as this!

So much content in `helpers.rs` that is better to be split across files.

Reply:
Done!